src/main/java/net/mcreator/ui/dialogs/workspace/GeneratorSelector.java

```diff
@@ -192,7 +208,24 @@ public static GeneratorConfiguration getGeneratorSelector(Window parent, @Nullab
 				JOptionPane.OK_CANCEL_OPTION, JOptionPane.PLAIN_MESSAGE);
 
 		if (resultval == JOptionPane.OK_OPTION && generator.getSelectedItem() != null) {
-			return (GeneratorConfiguration) generator.getSelectedItem();
+			if (workspace != null && parent.getParent() instanceof MCreator mcreator) {

```

Comment:
Why is here parent.getParent() and how is this universal?

Reply:
Also, now in case `workspace != null && parent.getParent() instanceof MCreator mcreator` is true, default logic will never run and always only work when switching from fabric to forge or vice versa.

Reply:
> Why is here parent.getParent() and how is this universal? What if the calee will be MCreator directly, or sub-sub-parent.

The generator selector dialog is always used after another dialog is opened, so when we create a new workspace or when we change it from the workspace settings. This is why I did this.

Reply:
Maybe add an additional optional parameter MCreator instead and overload this method with a variant where MCreator is null. Just idea, as the current approach is not really Javanic